src/Umbraco.Core/Migrations/Upgrade/V_8_7_0/ConvertToElements.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Linq;
+using Umbraco.Core.Persistence;
+using Umbraco.Core.Persistence.Dtos;
+
+namespace Umbraco.Core.Migrations.Upgrade.V_8_7_0
+{
+    public class ConvertToElements : MigrationBase
+    {
+        public ConvertToElements(IMigrationContext context) : base(context)
+        {
+        }
+
+        public override void Migrate()
+        {
+            // Get all document type IDs by alias
+            var docTypes = Database.Fetch<ContentTypeDto>();
```

Comment:
why not just do `Database.Fetch<ContentTypeDto>().ToDictionary(x => x.Alias, x => x.NodeId)` ?

Reply:
I had written an app a couple years back that did a lot of list to dictionary conversions, and it was performing horribly.  After doing using VS's performance analyzer, we found that the dictionary resize operations are extremely expensive, especially as the list grows to 1,000+ elements.  If you size the dictionary appropriately at the start and then just insert into it, it was lightning fast.  Ever since then, I've always tried to make it a habit to size the dictionary appropriately and then insert into it.  Since the ToDictionary extension is on the IEnumerable interface, it can't do that, and so just iterates and adds to the dictionary.  If they created a ToDictionary that was an extension on the ICollection interface, that would be ideal, but without that I just create my own, properly sized dictionary and add to it.

Reply:
Fair enough! Might be worth making an ext method for this then based on ICollection<T>, not required for this PR but in the future. And good to know! Wonder if there are places in our codebase where we are creating large dictionaries with ToDictionary based on collections, might have a look today and see.

Reply:
I decided to create a PR for changing some of our internal usages of ToDictionary and discovered we should be using ToLookup in various places too instead of GroupBy/ToDictionary which is much slower. I've left this open for the community to review but please have a look i fyou have time https://github.com/umbraco/Umbraco-CMS/pull/7969